StripeCore/StripeCore/Source/API Bindings/STPAPIClient.swift

```diff
@@ -400,9 +417,15 @@ extension STPAPIClient {
         }
 
         request.httpMethod = method.rawValue
-        for (k, v) in authorizationHeader(using: ephemeralKeySecret) {
+        for (k, v) in authorizationHeader(using: ephemeralKeySecret ?? consumerPublishableKey) {
             request.setValue(v, forHTTPHeaderField: k)
         }
+        
+        if consumerPublishableKey != nil {
+            // If we now have a consumer publishable key, we no longer send the connected account
+            // in the header, as otherwise the request will justifiably fail.
+            request.setValue(nil, forHTTPHeaderField: "Stripe-Account")
```

Comment:
Very minor, but it might be nice to pull this string out into a constant now that it's in this file twice